File: CPP/ListasEnlazadasCPP/Node.cpp
#include <iostream>
using namespace std;

template <typename E>

class Node{
    
    public:
        E data;
        Node<E>* next;
    
    public:
        //No se usa this.first = null;
        //Como es un puntero a un Objecto Node, toca ponerle null-pointer (nullptr)
        Node() : next(nullptr) {}

        Node(E data) : data(data), next(nullptr) {}
    
        Node(E data, Node<E>* next) : data(data), next(next) {}

        E getData(){
            return data;
        }

        void setData(E data){
            this->data = data;
        }

        Node<E>* getNext(){
            return next;
        }

        void setNext(Node<E>* next){
            this->next = next;
        }

        string toString(){
            //No se puede concatenar integers con cadenas, toca usar:
            //to_string(data);
           return "Node(data: " + to_string(data) + ")";
        }
};

File: 2024-1 (III Semestre)/Estructuras de Datos/Estr_DinamicGraph/src/simple/Node.java
package simple;
public class Node<E> {

	private E info;
	private Node<E> next;

	public Node(E info, Node<E> next) {
		this.info = info;
		this.next = next;
	}

	public Node(E info) {
		this(info, null);
	}

	public Node() {
		this(null, null);
	}

	public E getInfo() {
		return this.info;
	}

	public Node<E> getNext() {
		return this.next;
	}

	public void setInfo(E info) {
		this.info = info;
	}

	public void setNext(Node<E> next) {
		this.next = next;
	}

	public String toString() {
		if (info != null) {
			return info.toString();
		} else {
			return null;
		}
	}
}